### DISCUSSION TITLE
Alternative way to handle the query params

### DISCUSSION THREAD

**Discussion by noxify** [sadmann7/tablecn#184] [Ideas]
I played a bit around with your template and tried https://nuqs.47ng.com/ for the query param handling.

Would it be an option for the tempate to use nuqs instead of the current implementation?

If you need some inspiration, here it is: 
https://github.com/noxify/reactquerybuilder-nextjs-rsc-queryparams

Differences:

* The example uses prisma instead of drizzle 
  ( Reason: we use prisma internally in our project )
* I removed some stuff which wasn't relevant for me
* For pagination, I use the parameters which are expected from tanstack table
  Results in `pageIndex=0` instead of `page=1`
* for the filtering, I use https://react-querybuilder.js.org/ but shouldn't be a problem to integrate it in your implementation

What could be the benefits?

For me, it was the easier handling while implementating custom query params.
I don't have to take care about all the `useEffect` magic in the code.
( if you haven't mentioned, I removed most of them)

This results in shorter, for me, better maintainable code.

Since nuqs works also with RSC, there is no limitation.

If you want, I can provide a working with drizzle and the current filter implementation.

P.S: If you like the react querybuilder implementation for the advanced filter, feel free to copy ;)

**Comment by sadmann7** [comment]
Looks super clean. Will look through it.

**Reply by zbeyens** [reply]
Here is a quick migration of `useDataTable` to `nuqs` https://gist.github.com/zbeyens/9eb6f361fc3c1f1cb09ac3c93e1a6298
This allows to use replace history instead of push.

**Comment by dBianchii** [comment]
Amazing library I hadn't heard of before. I noticed that many times I felt that when using this table, I noticed that often the query params weren't being well reflected. It felt unresponsive and I noticed that sometimes when I removed one filter or swapped one to another, it would keep a dead nonexistent filter in the url. Many many cases.

Maybe if we used this nuqs library then the chances of these bugs happening would be lower bc it would be easier to atomatically update and maintain?

**Comment by nckre** [comment]
nuqs just released [v2](https://nuqs.47ng.com/blog/nuqs-2), includes a testing adapter for URL and a [community parser](https://nuqs.47ng.com/docs/parsers/community/tanstack-table) for TanstackTable.
